snforge_std/src/cheatcodes/contract_class.cairo

```diff
@@ -87,11 +94,33 @@ impl ContractClassImpl of ContractClassTrait {
     }
 }
 
+trait DeclareResultTrait {
+    fn contract_class(self: @DeclareResult) -> @ContractClass;
+
+    fn success_contract_class(self: @DeclareResult) -> @ContractClass;
+}
+
+impl DeclareResultImpl of DeclareResultTrait {
+    fn contract_class(self: @DeclareResult) -> @ContractClass {
+        match self {
+            DeclareResult::Success(contract_class) => contract_class,
+            DeclareResult::AlreadyDeclared(contract_class) => contract_class
+        }
+    }
+
+    fn success_contract_class(self: @DeclareResult) -> @ContractClass {
+        match self {
+            DeclareResult::Success(contract_class) => contract_class,
+            DeclareResult::AlreadyDeclared(_) => panic!("Class hash is already declared")
+        }
```

Comment:
Don't like this as a public interface, at least not named like this. It is not obvious it can panic. 

Either add `unwrap`/`assert` to the name or not include these at all. I'm not sure if there is relevant usecase for this method for end users.

Reply:
It's just an analogy to Rust's `Result` enum, so there definitely should be a method like this. I know the naming is unfortunate, but what about `unwrap_success`? The second method name could remain the same.

Reply:
I think `unwrap_sucess` is okay.

Reply:
Not sure I get the usecase for asserting/unwrapping if the contract got successfully declared. In both cases of DeclareResult, you are getting an information that the contract is declared and can be used, not sure in which cases would the user want to verify if it was not previously declared.

Reply:
After discussing this offline with other team member, we decided to remove this